src/khoj/processor/conversation/offline/utils.py

```diff
@@ -1,43 +1,59 @@
+import glob
 import logging
+import os
+
+from huggingface_hub.constants import HF_HUB_CACHE
 
 from khoj.utils import state
 
 logger = logging.getLogger(__name__)
 
 
-def download_model(model_name: str):
+def download_model(repo_id: str, filename: str = "*Q4_K_M.gguf"):
     try:
-        import gpt4all
+        from llama_cpp.llama import Llama
     except ModuleNotFoundError as e:
-        logger.info("There was an error importing GPT4All. Please run pip install gpt4all in order to install it.")
+        logger.info(
+            "There was an error importing llama-cpp-python. Please run pip install llama-cpp-python in order to install it."
```

Comment:
As this is going into the default packages used in `pyproject.toml`, this whole `try/except` block should be unnecessary, I believe. Installations would be caught when the package is first installed.